### DISCUSSION TITLE
psycopg2 connection remains in hung state during RDS failover

### DISCUSSION THREAD

**Discussion by roshanpathak500** [psycopg/psycopg2#1762] [Q&A]
I am trying to connect to postgres database (hosted on AWS RDS server) from python application .

The **issue** is whenever there is a RDS reboot with failover , the code gets stuck for an hour without throwing "SSL SYSCALL error: Operation timed out" exception . Not just this, the hung state blocks the main process on the EKS pod in which python application is running in turn not allowing any requests to be honored .
On the Contrary, when I simply shutdown the RDS (instead of failover) , the process immediately fails and shows the exception allowing me to do further operations .

```
from psycopg2.pool import SimpleConnectionPool

def get_psycopg2_conn():
        try:
            connection_pool = SimpleConnectionPool(
                minconn=5,
                maxconn=30,
                keepalives=1,
                keepalives_idle=30,
                keepalives_interval=10,
                keepalives_count=5,
                host=“myhost”,
                database=“myappdb”,
                user=“dummyuser”,
                password=“dummypassword”
            )
            return connection_pool.getconn()
        except:
            return None
```
Removing any pool strategy also does not work . 

```
 conn = psycopg2.connect(
                host="myhost"
                database="myappdb",
                user="dummyuser",
                password=""dummypassword"            
                )
```

What can be done to ensure the execution does not go into hung state rather throws exception right away ?

**Comment by dvarrazzo** [comment]
We cannot support RDS for free: we cannot run any test for free, as it is a paid-for service.

If you want to purchase consultancy time feel free to reach out privately.
